tutorials/2022-02-09-neow3j-nep11/index.md

```diff
@@ -0,0 +1,647 @@
+
+Adds the provided key-value pair information in the manifest's `extra` field. You can also use `@ManifestsExtras` to gather multiple `@ManifestExtra` annotations (results in the same as when using single `@ManifestExtra` annotations).
+
+_`@SupportedStandards`_
+
+Sets the `supportedStandards` field in the manifest.
+
+```java
+@DisplayName("FurryFriends")
+@ManifestExtra(key = "author", value = "AxLabs")
+@SupportedStandards("NEP-11")
+public class NonFungibleToken {
+```
+
+### Constants
+
+You can set a constant value for the contract by using `final` variables. These values are always loaded when the contract is called and cannot be changed once the contract is deployed.
+
+> Note: The contract owner of this example contract is fixed (i.e., it is a `final` variable). If you intend to provide a functionality to change such a variable, you should not store it as a `final` variable. Rather, you would store it as a value in the storage, that you then can change through a method.
+
+-
+
+> Note: All contract constants and all methods have to be `static` (since the object-orientation of the JVM is different on the NeoVM).
+
+-
+
+```java
+    static final Hash160 contractOwner = StringLiteralHelper.addressToScriptHash("NM7Aky765FG8NhhwtxjXRx7jEL1cnw7PBP");
+
+    static final StorageContext ctx = Storage.getStorageContext();
+    static final StorageMap contractMap = new StorageMap(ctx, 0);
+    static final StorageMap registryMap = new StorageMap(ctx, 1);
+    static final StorageMap ownerOfMap = new StorageMap(ctx, 2);
+    static final StorageMap balanceMap = new StorageMap(ctx, 3);
+
+    // Keys of key-value pairs in NFT properties
+    static final String propName = "name";
+    static final String propDescription = "description";
+    static final String propImage = "image";
+    static final String propTokenURI = "tokenURI";
+
+    static final StorageMap propertiesNameMap = new StorageMap(ctx, 8);
+    static final StorageMap propertiesDescriptionMap = new StorageMap(ctx, 9);
+    static final StorageMap propertiesImageMap = new StorageMap(ctx, 10);
+    static final StorageMap propertiesTokenURIMap = new StorageMap(ctx, 11);
+
+    static final byte[] totalSupplyKey = new byte[]{0x10};
+    static final byte[] tokensOfKey = new byte[]{0x11};
+```
+
+### NEP-11 Methods
+
+The required NEP-11 methods are implemented in the following. If a method does not change the state of the contract (i.e., it is just used for reading), it can be annotated with the `@Safe` annotation. Out of the required NEP-11 methods, only the `transfer` method should be writing to the contract and is thus not annotated as safe.
+
+> Note: Java's `assert` statement is used in neow3j to throw exceptions on the NeoVM. Thus, there is no need of using an `if` statement with a followed `throw new Exception()` statement. Although, it may still be used like that. It is up to the developer's preference, as it results in the exact same NeoVM code once compiled.
```

Comment:
```suggestion
> Note: Java's `assert` statement is used in neow3j to throw exceptions on the NeoVM. Thus, there is no need of using an `if` statement with a followed `throw new Exception()` statement. Although, it may still be used like that. It is up to the developer to go either way, as it results in the exact same NeoVM code once compiled.
```